pkg/diagnose/scan.go

```diff
@@ -44,43 +45,13 @@ func Scan(ctx context.Context, opts ...OpOption) error {
 	}
 
 	fmt.Printf("\n\n%s scanning the host (GOOS %s)\n\n", inProgress, runtime.GOOS)
-	machineID, err := host.GetMachineID(ctx)
-	if err != nil {
-		log.Logger.Warnw("error reading machine ID", "error", err)
-	} else {
-		fmt.Printf("%s detected machine ID %q\n", checkMark, machineID)
-	}
-
-	bootID, err := host.GetBootID()
-	if err != nil {
-		log.Logger.Warnw("error reading boot ID", "error", err)
-	} else {
-		fmt.Printf("%s detected boot ID %q\n", checkMark, bootID)
-	}
-
-	virtEnv, err := host.GetSystemdDetectVirt(ctx)
-	if err != nil {
-		log.Logger.Warnw("error detecting virtualization environment", "error", err)
-	} else {
-		fmt.Printf("%s detected virtualization environment %q\n", checkMark, virtEnv.Type)
-	}
-	manufacturer, err := host.GetSystemManufacturer(ctx)
-	if err != nil {
-		log.Logger.Warnw("error detecting system manufacturer", "error", err)
-	} else {
-		fmt.Printf("%s detected system manufacturer %q\n", checkMark, manufacturer)
-	}
 
-	fmt.Printf("%s scanning the process counts\n", inProgress)
-	processCountsByStatus, err := process.CountProcessesByStatus(ctx)
+	osData, err := oscomponent.CollectData(ctx)
 	if err != nil {
-		log.Logger.Warnw("error counting processes by status", "error", err)
-	} else {
-		fmt.Printf("%s counted processes by status\n", checkMark)
-		for status, count := range processCountsByStatus {
-			fmt.Printf("%s %q %d\n", checkMark, status, len(count))
-		}
+		return err
 	}
+	fmt.Printf("%s %s\n", osData.HealthState().TerminalHeader(), osData.Summary())
+	fmt.Println(osData.String())
```

Comment:
Shall we print the information without osData? The information is static almostly except that process counts.  And after gossip refactoring,  the os package only update some metrics.

Reply:
The intention is, summary + Health state methods represent much condensed information about health state. And `String` just emits all data (not all, but important information), as it's used for "scanning"